### ISSUE TITLE
Homebridge never adds on iOS 11

### ISSUE THREAD

**Issue by AutumnForesty** [nfarina/homebridge#1514]
I recently set up homebridge again on antergos, and it starts up fine.

It's adding it into the home app that's the issue.

I go and add it, and after some time, the home app says "Home couldn't connect to this accessory."
I'm on the same wifi network as the bridge as well.
What could be the issue here?

I'm running the latest iOS 11, and the latest homebridge.

**Comment by erpeters3** [comment]
Exact same issue.  I was having connectivity issues, random drops.  I removed the bridge and attempted to re-add on iOS11 GM, no joy.

**Comment by qudarr** [comment]
ditto for me. Used to work with IOS 11 Beta, now very unreliable connectivity.

**Comment by erpeters3** [comment]
Now i cannot re-add my Homebridge even back on iOS 10.3.3.  Updated Pi Jessie, NPM, Homebridge, and Homebridge plugins to latest.  No Home apps on any iOS device see them.  Please help before my wife divorces me ;)

**Comment by apetitjean** [comment]
Same issue here! My wife and I have 2 iPhones. I updated mine to iOS 11 and since then I cannot connect to homebridge anymore :-(.
My wife who's still running iOS 10.x has no problem...

**Comment by erpeters3** [comment]
Following this advice from issue #1501 fixed it had to rebuild everything but st least it works

What I did which worked for me was:
	•	Upgrade homebridge to newest version: npm install -g homebridge
	•	Stop homebridge service
	•	rm -rf ~/.homebridge/persist
	•	rm -rf ~/.homebridge/accessories
	•	Under ~/.homebridge/config.json, change your "username": "11:22:33:44:55:66" to something else
	•	Start homebridge service
	•	Delete homebridge accessory from home app
	•	Add homebridge accessory to home app and enter pin etc

**Comment by AutumnForesty** [comment]
@erpeters3 

Did that, no dice.

**Comment by jcastro** [comment]
@erpeters3 @AutumnForesty not working here either :( but thanks!

**Comment by ttscoff** [comment]
The solution from @erpeters3 worked for me. Thanks!

I still haven't figured out how to _delete_ the old items from Home in iOS 11, but one step at a time.

**Comment by jg506** [comment]
FWIW, I got iOS11 issue resolved only after adding following lines to config.json which I previously  omitted. It seems like one or more of these are required by iOS 11 but not required by iOS 10.
        "manufacturer": "@nfarina",
         "model": "Homebridge",
         "serialNumber": "0.4.20"

**Comment by jcastro** [comment]
Thanks @jg506, I just tried but still I get Homebridge running perfectly but I can't find the accessory on my iPhone 7 with iOS11 :(. I've tried everything I could find but no luck. It's strange because with iOS10 and previous versions of Homebridge it was working just fine

**Comment by erpeters3** [comment]
And you’re absolutely certain you used nano to change your hex username to something new and properly saved the config.json file?

**Comment by jcastro** [comment]
Yep, I'm using Atom and making sure the correct file type is selected (on this case JSON)

**Comment by PencilDH** [comment]
I have updated my My apple stuff to iOS 11, had issues with communicating with all home bridge accessories...

I didn't delate the persist folder (I was lazy as I have all devices and rooms reconfigured in home app and this take me at least 20 minutes to set back up is I would like to avoid this) , so I just added:

`   
    "bridge": {
        "name": "Homebridge",
        "username": "CC:22:3D:E3:CE:30",
        "port": 51826,
        "pin": "031-45-154",
        **"manufacturer": "@nfarina",
         "model": "Homebridge",
         "serialNumber": "0.4.20"**
    },
`
to the config file and BOOM all good!

**Comment by apetitjean** [comment]
Hi all!

To make the Home App on iOS 11 work again after the OS upgrade, I just deleted the persist folder and reconfigured from scratch the Home App on my iPhone. I mean I had to delete all the app settings and restart the whole process and it just worked.
I'm pretty sure deleted the persist folder wasn't mandatory...

-Arnaud

**Comment by nfarina** [comment]
Sorry for the late chime-in here, but mine's been working fine on iOS 11 - I don't have any of the `manufacturer` `serialNumber` etc that @PencilDH added - I'm guessing that adding that had the side-effect of avoiding some cache issue with a previously-added bridge, and that may have fixed the problem.

**Comment by jg506** [comment]
@nfarina I must have deleted the persistence and accessories directories 10-20 or times , restarted 30 times, reinstalled completely using latest version 4-6 times, followed all advises to change name, username - and I still was not unable to join the bridge on iOS 11 (discovery worked, but failed at accessory does not respond after entering the code). Only adding these lines to config solved problem and no additional troubleshooting step was done concurrently.  So I would speculate these lines do matter on their own, but probably not at each and every situation (e.g. I have mix of iOS 10 and iOS11 devices with sharing across icloud accounts).

**Comment by B1tMaster** [comment]
I have upgraded from IOS 10 to IOS 11 on the fist day. Everything was working perfectly. Upgraded IOS 11 to latest version 2 days ago. Still everything was perfect. Today I had to shutdown my Raspberry PI and restart.. After that, my Home app can no longer see HomeBridge.  

Did everything described above and even deleted everything in my Home app.  (deleted home). Still can not see homebridge anymore.. 

This is most bizarre situation as it seem to be affecting random people with random setups..

**Comment by HSkul** [comment]
I have somewhat similar situation running Homebridge on a Raspberry Pi 1B.

If I remove 'persist' and 'accessories', and advance the user name by one digit, when I start Homebridge (with 'DEBUG=* homebridge -D') I can get my iOS 11 iPhone to pair with it.   However, my ATV4 does not appear to communicate with Homebridge at all (only my iPhone).  Which means that everything is fine until my iPhone disconnects from the wifi and I get

`Client connection error: read EHOSTUNREACH`

  If I then stop Homebridge, delete it from the Home app (but leave 'persist' and 'accessories' as is), start it again and pair it with the iPhone, my ATV4 starts communicating with Homebridge along with my phone.  However, it always ends up with it and the ATV disconnecting via

` EventedHTTPServer [::ffff:192.168.1.239] HTTP Response is finished +10ms`
 ` EventedHTTPServer [::ffff:192.168.1.244] Client connection closed +37s`
`  EventedHTTPServer [::ffff:192.168.1.244] HTTP connection was closed +4ms`
`  EventedHTTPServer [::ffff:192.168.1.244] HTTP server was closed +2ms`

  where 239 is my phone and 244 is the ATV.   The ATV disconnects first (phone keeps communicating) and often (not sure if always) the phone disconnects later in the same way.  If it doesn't disconnect it will loose connection when it disconnects from wifi.  But Homebridge keeps running fine and detects my garage door opening and closing.  

  I had it working for a while and then (idiot me) I upgraded my Raspberry Pi to Stretch and both the phone and ATV to the latest updates.    I think my network is fine as the Home app can always access my Ecobee thermostats just fine.

  Is there a well behaved simple plugin that everyone can use to check the connection issues?  Something that is platform independent?

**Comment by B1tMaster** [comment]
Problem solved: https://github.com/B1tMaster/haas-config/issues/54
This is the best way to solve it as it is pretty much guaranteed to work, since you would be installing the homebridge plugin via Docker and your environment, libraries , settings etc will be same regardless of what OS you run.

**Comment by qudarr** [comment]
Sorry, but I am not going to switch to a different platform.
All of the above mentioned tips didn't work, including:

- removing the accessories and persist directories
- incrementing the username
- restarting all devices
- upgrading npm and homebridge
- deleting the homebridge accessory from home app

Homebridge will startup fine, but not appear on any iOS 11 devices.

**Comment by B1tMaster** [comment]
@qudarr , as I said, I had all same issues. Nothing worked. all the steps you mentioned. 
But you if you follow my directions, it something you did not try.  At the end, you do what you wish mate.

**Comment by jg506** [comment]
> But you if you follow my directions, it something you did not try.
@B1tMaster -- you make it sound like it's a simple troubleshooting step but your proposal is quite extreme and effort intensive, considering all the implicit requirements of moving from one hub into another and in case of haas hub having to build hub on your own including setting up RPi putting zigbee and zwave radios in it, hoping that all your currently working devices and automations are transferable, learning this new platform, excluding and re-pairing all your devices and recreating automatons one by one.  

> hass.io uses docker and when we deploy software using Docker we are guaranteed
> to have exactly the same environment and versions of libraries and every other configuration setting as the developer of the plugin.

Good point, but you do not have to move over to haas.io just so you can build and run homebridge using docker. Many of us (myself included) are using dockerized homebridge setups without haas.io. Docker hub has many of these (just search there for homebridge). Anyone is welcome to use my build that does work w/ iOS11.

**Comment by B1tMaster** [comment]
@jg506 I agree with your sentiment. It is not a small change. Took me a better part of my weekend. But I actually wanted to find a good reason to migrate to haasio for future ease of upgrades anyway. I am happy with the effort spent, but it is by no means a 5 mins procedure.

**Comment by B1tMaster** [comment]
@jg506 "excluding and re-pairing all your devices and recreating automatons one by one."
Did not need to do any of that..  Your Z-wave network stays as is.. But I did have to go and "rename" all zwave nodes again.
 
All scripts and automations transferred  asis..  And it is much simpler to transfer code now when you install plugins like SAMBA.. 

The only real time killer was burning the new image for my SDcard, and figuring out the differences in operation of the new platform.. And a few other minor things  like had to rename "automation.yaml" to "automations.yaml" and script.yaml to scripts.yaml.   

I hope everybody will find a good way to resolve this.. For me the fact is that homebridge is now just a plug-in to HA is brilliant. NO need to worry about Homebridge starting up too early on reboot or deal manually with the upgrade. Not even dealing with homebridge config file as it is being generated automatically by plugin. I only had to update my password inside the plugin.

**Comment by aguaman15** [comment]
I combined what @erpeters3 and @jg506 did above and it worked. Thank you! :)

**Comment by AutumnForesty** [comment]
Solved my issue!

UFW (firewall) had the port blocked by default, and simply creating a rule fixed it for me. So for anyone else having issues, check your firewall settings on the PC / Mac / Linux machine too, as that could be stopping homebridge from talking to your iPhone, iPad and apple TV.  :)

**Comment by LeoCal** [comment]
Folks, in the end I managed to get everything working again. No need to run homebridge in Docker or to install Hass.io. What you need to do is a cleanup of your PI.
I basically removed all the packages (nodejs, npm, node-server, nodered, hap-nodejs, homebridge) and installed everything from scratch following the installing procedures at:
- https://github.com/KhaosT/HAP-NodeJS
- https://github.com/nfarina/homebridge
Special thanks to the support from @KhaosT and @nfarina while I was trying to troubleshoot their code (in the end their code was just fine :-)).

**Comment by B1tMaster** [comment]
@LeoCal,
Great to hear you solved it. But how is everything you described less work than moving to haas.io once and never having to go thru this pain again?

Haas.io is just: 
1.burn image and boot
2. Rename your zwave devices
3. Move ur original config files as is to new installation 
4. Install homebride plugin.

As a side benefit u get one click upgrade going forward, out of the box working upgradable w a click homebridge. And other things like automatic backup, samba plugin mount and edit ur config files from laptop, etc, etc..

**Comment by LeoCal** [comment]
@B1tMaster 
This is definitely true for someone who's approaching home automation and has to deploy a new system, however that doesn't work for me. I've built my own system over the years, customising it based on my needs.
I've had a look at Hass.io and, while I find it very interesting, I believe it doesn't really match what I'm looking for (yet) - I'll keep an eye on it and who knows, maybe one day I'll decide to switch to it.
I just find it odd having to move to a completely new Home Automation system just for one app (i.e. using Siri via Homebridge), when the problem lies in the fact that my system wasn't clean.
But thanks to your input I came to know about Hass.io and I'll monitor it to understand when it's the right moment to switch.
